OneCommunityGlobal/HighestGoodNetworkApp
# yihan create way to see 6 months & yearly anniversaries

# Description
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/3a13bf37-0d27-477e-bed9-abb10b51766d)

## Related PRS (if any):
This frontend PR is related to the [#627](https://github.com/OneCommunityGlobal/HGNRest/pull/627) backend PR.

## Main changes explained:
- Update file leaderBoardData.js, URL.js and LeaderBoard.container.jsx for including a function of posting trophy data.
- Update file leaderBoard.jsx and FormattedReport.jsx for adding trophy icon to leaderboard and summary page.
- Create file anniversaryPermissions.js for date calculation functions.

## How to test:
1. check into current branch
2. do `npm run start:local` to run this PR locally
3. log as owner/admin/core/mentor/manager user
4. go to leaderboard, verify if you can see a trophy icon with 6M, 1Y, ..., 10Y under a user's name whose anniversary is in the past week. 
a. For example, today is Nov 29, so the trophy icon is showed if the create date of the user if between Nov 23 and Nov 29.
5. Click on the trophy icon-> Confirm botton, verify if the trophy icon then turns yellow. (This process may take some time)
6. Go to Reports → Weekly Summaries Reports Page, verify if you can see the trophy icon of the user whose create date is in this week. 
a. For example, today's date is Nov 29, so under "This week" tab, the trophy icon is showed if create date is between Nov 26 and Dec 2, under "Last Week" tab, the icon is showed if create date is between Nov 19 and Nov 25... 

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/94303659/f7e23e2d-85c5-4f4c-9a90-a81bda60d89a
